### ISSUE TITLE
Make Block Post Type accessible in Post Object - Feature Request

### ISSUE THREAD

**Issue by riladant** [AdvancedCustomFields/acf#196]
Gutenberg reusable blocks are saved as their own post type (wp_block) accessible from /wp-admin/edit.php?post_type=wp_block

Can this post type be added to the 'Post Object' field type?

When used in conjunction with a repeater field, this would make it very easy to implement reusable content

![Screen Shot 2019-06-25 at 11 17 42 am](https://user-images.githubusercontent.com/16073729/60063222-5b2c3800-973b-11e9-970b-1b06d5026d4c.png)

**Comment by elliotcondon** [comment]
Hi @riladant 

thanks for the feature request. 

Currently, we only allow "public" post types to be selectable for these kinds of settings, but I do like the idea of adding "wp_block" to the list.

For now, please hook into the "acf/get_post_types" filter and add in "wp_block" to the array of post types :)

**Comment by riladant** [comment]
In case others are looking to do this in the interim, add this to your `functions.php`

```php
// ADD wp_block POST TYPE TO ACF POST OPTIONS
function lh_filter_acf_get_post_types( $post_types ) {
  if(!in_array('wp_block', $post_types)){
    $post_types[] = 'wp_block';
  }
  return $post_types;
};
add_filter( 'acf/get_post_types', 'lh_filter_acf_get_post_types', 10, 1 );
```

**Comment by FriendlyWP** [comment]
Another option is to make the Reusable Blocks public. You can do that using this function.  This also allows for Reusable Block previews if you have made those visible / editable in the control panel ([see how here](https://github.com/billerickson/Reusable-Blocks-UI)).

```
// Enable Previews for Reusable Block 
add_filter( 'register_post_type_args', 'change_post_type_args', 10, 2 );
function change_post_type_args( $args, $post_name ) {
    if ($post_name == 'wp_block') {
        $args['public'] = true;
        $args['publicly_queryable'] = true;
        $args['rewrite'] = true;
    }
    return $args;
}
```

**Comment by robwent** [comment]
@FriendlyWP bear in mind that that will also add the post type to sitemaps and allow them to be indexed by search engines which may not be a desirable result.

**Comment by FriendlyWP** [comment]
@robwent True, but it's easy to get around that if you're using any SEO plugin (Yoast SEO, etc). Personally I think being able to use these with ACF and preview the blocks makes it worth the risk.

**Comment by robwent** [comment]
@FriendlyWP there is a benefit. I just wanted to point it out for anyone finding this through Google.

**Comment by robwent** [comment]
@FriendlyWP I haven't looked through the plugin code but I would guess that you could modify your snippet to check if a user is logged in to do a similar thing.
Then you would get the preview but the post type still wouldn't be inaccessible to guest users.
